torch/_inductor/decomposition.py

```diff
@@ -74,225 +74,22 @@ def empty_permuted(size, physical_layout, **kwargs):
-    else:
-        return 0
+
 
 
 def check_device(a: Tensor, b: Tensor):
     return a.is_cuda and b.is_cuda
 
 
-def get_padded_length(x, alignment_size):
-    if alignment_size == 0 or x % alignment_size == 0:
-        return 0
-    return int((x // alignment_size + 1) * alignment_size) - x
-
-
-def pad_dim(x, padded_length, dim):
-    if padded_length == 0:
-        return x
-    pad = x.new_zeros(*x.shape[:dim], padded_length, *x.shape[dim + 1 :])
-    return torch.cat([x, pad], dim=dim)
-
-
-@register_decomposition([aten.addmm])
-def addmm(input, mat1, mat2, *, beta=1, alpha=1):
-    if (
-        config.shape_padding
-        and check_device(mat1, mat2)
-        and should_pad_bench(mat1, mat2, torch.ops.aten.addmm, input=input)
-    ):
-        m_padded_length = get_padded_length(mat1.shape[0], get_alignment_size(mat1))
-        k_padded_length = get_padded_length(mat1.shape[1], get_alignment_size(mat1))
-        n_padded_length = get_padded_length(mat2.shape[1], get_alignment_size(mat2))
-        if m_padded_length != 0 or k_padded_length != 0 or n_padded_length != 0:
-            return pad_addmm(
-                input, mat1, mat2, m_padded_length, k_padded_length, n_padded_length
-            )
-
-    return NotImplemented  # go directly to lowering
-
-
-def pad_addmm(input, mat1, mat2, m_padded_length, k_padded_length, n_padded_length):
-    # addmm decomp with padding will go through pad_addmm multiple times if multiple dimensions are needed to be padded
-    if k_padded_length != 0:
-        mat1 = pad_dim(mat1, k_padded_length, 1)
-        mat2 = pad_dim(mat2, k_padded_length, 0)
-    elif n_padded_length != 0:
-        mat2 = pad_dim(mat2, n_padded_length, 1)
-    elif m_padded_length != 0:
-        mat1 = pad_dim(mat1, m_padded_length, 0)
-
-    if input is not None and k_padded_length == 0:
-        if n_padded_length != 0:
-            if input.dim() == 2:
-                input = pad_dim(input, n_padded_length, 1)
-            elif input.dim() == 1:
-                input = pad_dim(input, n_padded_length, 0)
-        elif m_padded_length != 0 and input.dim() == 2:
-            input = pad_dim(input, m_padded_length, 0)
-
-    if k_padded_length != 0:
-        return torch.ops.aten.addmm(input, mat1, mat2)
-    elif n_padded_length != 0:
-        return torch.ops.aten.addmm(input, mat1, mat2)[:, :-n_padded_length]
-    else:
-        return torch.ops.aten.addmm(input, mat1, mat2)[:-m_padded_length, :]
-
-
-def should_pad_bench(mat1, mat2, op, input=None):
-    assert utils.has_triton()
-    from triton.testing import do_bench
-
-    with no_dispatch():
-        if op is torch.ops.aten.mm or op is torch.ops.aten.addmm:
-            m_padded_length = get_padded_length(mat1.shape[0], get_alignment_size(mat1))
-            k_padded_length = get_padded_length(mat1.shape[1], get_alignment_size(mat1))
-            n_padded_length = get_padded_length(mat2.shape[1], get_alignment_size(mat2))
-        elif op is torch.ops.aten.bmm:
```

Comment:
For now I only handle `addmm` but will need to also look at `mm` and `bmm`

At least for linear layers what i seemed to be getting was `addmm`

Reply:
Probably shouldn't delete these until your pass actually works for all these operators.